MantidQt/MantidWidgets/src/InstrumentView/PanelsSurface.cpp

```diff
@@ -231,9 +219,8 @@ void PanelsSurface::addFlatBank(ComponentID bankId,
     m_unwrappedDetectors.reserve(m_unwrappedDetectors.size() + nelem);
     for (int i = 0; i < nelem; ++i) {
       // setup detector info
-      Mantid::Geometry::IDetector_const_sptr det =
-          boost::dynamic_pointer_cast<const Mantid::Geometry::IDetector>(
-              assembly->getChild(i));
+      auto det = boost::dynamic_pointer_cast<const Mantid::Geometry::IDetector>(
+          assembly->getChild(i));
       if (!doneRotation) {
         pos0 = det->getPos();
```

Comment:
I understand that you are iterating over a bank here, but we should not get positions or rotations from the detector directly. Maybe we could extract the `detector ids` from the detector, create a list of these and then use `DetectorInfo::indexOf` to find the index to get the position for? Maybe this would be a suitable half-way solution until we have some kind of `ComponentInfo`. @SimonHeybrock what do you think?